packages/date-picker/src/vaadin-date-picker-overlay-content.js

```diff
@@ -309,15 +243,11 @@ class DatePickerOverlayContent extends ControllerMixin(ThemableMixin(DirMixin(Po
    * @private
    */
   get __useSubMonthScrolling() {
-    return this.$.monthScroller.clientHeight < this.$.monthScroller.itemHeight + this.$.monthScroller.bufferOffset;
-  }
-
-  get calendars() {
-    return [...this.shadowRoot.querySelectorAll('vaadin-month-calendar')];
+    return this._monthScroller.clientHeight < this._monthScroller.itemHeight + this._monthScroller.bufferOffset;
   }
 
   get focusableDateElement() {
-    return this.calendars.map((calendar) => calendar.focusableDateElement).find(Boolean);
+    return this.calendars.map((calendar) => calendar.isConnected && calendar.focusableDateElement).find(Boolean);
```

Comment:
Added `isConnected` because some tests that open overlay might exit yearly, without waiting for date element to be focused. So, after the promise that waits for them to render is resolved, this getter is called, and the error is thrown on accessing `vaadin-month-calendar` shadow DOM, because the calendar is already detached.

I will try to fix those tests instead of using this workaround (although we could keep it as well).

Reply:
Fixed some tests and removed `isConnected` check, it should not be needed anymore.